Cargo.toml

```diff
@@ -5,21 +5,25 @@ edition = "2021"
 license = "MIT OR Apache-2.0"
 
 [features]
+default = ["stream"]
 expose_stream = []
+stream = ["dep:rustls", "dep:tokio", "dep:tokio-rustls"]
```

Comment:
Do you plan to integrate `async-std` or `native-tls` support inside the lib? If not (which I could understand), then `stream` is fine, but if so then `stream` sounds too generic. Sth ugglier but more explicit like `tokio-rustls` leaves room for later `tokio-native-tls` or `async-std-rustls` features.

Reply:
Then we would need to rename the `stream` module as well because features are additive. In this PR I tried to reuse existing files and types to reduce clutter. But it would totally make sense to have different stream implementations in this repo in the future (but this is currently low priority).

I'm not sure about the general direction for this. Should we use feature flags? Should we use separate crates, e.g. `imap-stream-tokio-rustls`, `imap-stream-async-std-rustls`, etc.

I would suggest to keep the names for now. And I'll create an issue for discussing this topic.

Reply:
I had the same thought. I think we could at least use "tokio", " async-std", and "sync" (or something similar). Not sure about the TLS part. Having all combinations is kinda... ugh... Maybe we can postpone this question a bit, use the simpler feature names, and decide at a later time?

Reply:
The types could also have the same names. I can't really imagine anyone using `tokio` and `async-std` at the same time. If so, the imports would require an `Stream as TokioStream` or so.

Reply:
Reading what jakob said... Let's open an issue as a reminder and postpone this question until we collected more experience.

Reply:
`Stream as TokioStream` pattern makes most sense to me. A module hierarchy like:

```text
root
└── stream
    ├── async_std
    │   └── mod.rs
    ├── mod.rs
    ├── stream_trait.rs
    └── tokio
        └── mod.rs
```

Reply:
I created issue #169